From 82a4725015d9e0e31544c0ab853ef8f2889c53f2 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Wed, 13 Feb 2013 12:10:41 -0600
Subject: [PATCH 671/691] ENGR00244769-3 [NOR FLASH]-Improve WEIM NOR speed

Improvements for WEIM NOR read performance.
Changed burst and page size to 8 words.
Changed read wait states to 10 and page read wait states to 4.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 board/freescale/mx6q_sabreauto/mx6q_sabreauto.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/board/freescale/mx6q_sabreauto/mx6q_sabreauto.c b/board/freescale/mx6q_sabreauto/mx6q_sabreauto.c
index 1f6902f..0a53dd9 100644
--- a/board/freescale/mx6q_sabreauto/mx6q_sabreauto.c
+++ b/board/freescale/mx6q_sabreauto/mx6q_sabreauto.c
@@ -723,10 +723,10 @@ iomux_v3_cfg_t nor_pads[] = {
 static void weim_norflash_cs_setup(void)
 {
     writel(0x00000120, WEIM_BASE_ADDR + 0x090);
-    writel(0x00620181, WEIM_BASE_ADDR + 0x000);
+    writel(0x00020181, WEIM_BASE_ADDR + 0x000);
     writel(0x00000001, WEIM_BASE_ADDR + 0x004);
-    writel(0x0f020000, WEIM_BASE_ADDR + 0x008);
-    writel(0x0000b000, WEIM_BASE_ADDR + 0x00c);
+    writel(0x0a020000, WEIM_BASE_ADDR + 0x008);
+    writel(0x0000c000, WEIM_BASE_ADDR + 0x00c);
     writel(0x0804a240, WEIM_BASE_ADDR + 0x010);
 }
 
-- 
1.7.7.4

